File: JavaScript/fibonacci.js
// Função que calcula o n-ésimo número de Fibonacci de forma recursiva
function fibonacci(a){
    // Se a entrada for 1 ou 0, retorna 1 (caso base)
    if(a == 1 || a == 0){
        return 1;
    }else{
        // Retorna a soma dos dois números anteriores na sequência de Fibonacci
        return (fibonacci(a-1) + fibonacci(a-2));
    }
}

// Lê a entrada do usuário e a armazena na variável 'fs'
fs = require("fs").readFileSync(0, "utf8");
// Converte a entrada para um número inteiro
fs = parseInt(fs);

// Chama a função Fibonacci com a entrada e exibe o resultado no console
console.log(fibonacci(fs));

File: C/fibonacci.c
#include <stdio.h>

// Função recursiva que calcula o n-ésimo termo da sequência de Fibonacci
// A sequência de Fibonacci é definida por: 
// F(0) = 1, F(1) = 1 e F(n) = F(n-1) + F(n-2) para n > 1.
int fibo(int n){
    // Caso base: se n for 0 ou 1, retorna 1, pois F(0) = F(1) = 1.
    if(n == 0 || n == 1){
        return 1;
    }else{
        // Chama recursivamente a função para calcular os dois termos anteriores
        return fibo(n-1) + fibo(n-2);
    }
}

int main(){
    int n, fi;
    
    // Lê o valor de 'n' (a posição do termo da sequência de Fibonacci)
    scanf("%d", &n);
    
    // Chama a função 'fibo' para calcular o n-ésimo termo
    fi = fibo(n);
    
    // Imprime o n-ésimo termo da sequência de Fibonacci
    printf("%d", fi);
    
    return 0;
}
